DOCS-6043-HTTP-responseTimeout-kt (#715)
mulesoft/docs-connectors: modules/ROOT/pages/http/http-documentation.adoc

HTTP Connector v1.5

> Anypoint Connector for HTTP (HTTP Connector) is used to handle and perform HTTP requests. This class

Reviewer: ```suggestion
Use Anypoint Connector for HTTP (HTTP Connector) to handle and perform HTTP requests. This class
```

** ALWAYS
** NEVER |  Whether to specify a correlationId when publishing messages. This applies both for custom correlation ids specifies at the operation level and for default correlation Ids taken from the current event |  AUTO |
| Preserve Headers Case a| Boolean |  By default, header keys are stored internally in lower-case. This is to improve performance of headers handling and is functionally correct as specified in the RFC. <p> In the case a server expects headers in a specific case, this flag may be set to true so the case of the header keys are preserved. |  false |

> | Response Timeout a| Number |  Maximum time in milliseconds that the request element will block the execution of the flow waiting for the HTTP response. If this value is not present, the default response timeout from the Mule configuration will be used, which is also 10000 | 10000 |

Reviewer: ```suggestion
| Response Timeout a| Number |  Maximum time in milliseconds that the request element blocks the execution of the flow waiting for the HTTP response. If this value is not present, the connector uses the default response timeout from the Mule configuration, which is also 10000 | 10000 |
```
Please check if this is accurate.

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Path a| String | The location (which will be resolved relative to the current classpath and file system, if possible) of the trust store. |  |
| Password a| String | The password used to protect the trust store. |  |
| Type a| String | The type of store used. |  |
| Algorithm a| String | The algorithm used by the trust store. |  |

> | Insecure a| Boolean | If true, no certificate validations will be performed, rendering connections vulnerable to attacks. Use at your own risk. |  |

Reviewer: ```suggestion
| Insecure a| Boolean | If true, no certificate validations are performed, rendering connections vulnerable to attacks. Use at your own risk. |  |
```

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required

> | Path a| String | The location (which will be resolved relative to the current classpath and file system, if possible) of the key store. |  |

Reviewer: ```suggestion
| Path a| String | The location (which resolves relative to the current classpath and file system, if possible) of the key store. |  |
```

| Type a| String | The type of store used. |  |

> | Alias a| String | When the key store contains many private keys, this attribute indicates the alias of the key that should be used. If not defined, the first key in the file will be used by default. |  |

Reviewer: ```suggestion
| Alias a| String | When the key store contains many private keys, this attribute indicates the alias of the key that should be used. If not defined, the first key in the file is used by default. |  |
```

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required

> | Fails Deployment a| Boolean | When the application is deployed, a connectivity test is performed on all connectors. If set to true, deployment will fail if the test doesn't pass after exhausting the associated reconnection strategy |  |

Reviewer: ```suggestion
| Fails Deployment a| Boolean | When the application is deployed, a connectivity test is performed on all connectors. If set to true, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy |  |
```

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required

> | Initial Buffer Size a| Number | This is the amount of memory that will be allocated in order to consume the stream and provide random access to it. If the stream contains more data than can be fit into this buffer, then it expands according to the bufferSizeIncrement attribute, with an upper limit of maxInMemorySize. |  |

Reviewer: ```suggestion
| Initial Buffer Size a| Number | This is the amount of memory allocated in order to consume the stream and provide random access to it. If the stream contains more data than can be fit into this buffer, then the buffer expands according to the bufferSizeIncrement attribute, with an upper limit of maxInMemorySize. |  |
```

> | Buffer Size Increment a| Number | How much the buffer size expands if it exceeds its initial size. Setting a value of zero or lower will mean that the buffer should not expand, meaning that a STREAM_MAXIMUM_SIZE_EXCEEDED error will be raised when the buffer gets full. |  |

Reviewer: ```suggestion
| Buffer Size Increment a| Number | Specifies how much the buffer size expands if the buffer exceeds its initial size. Setting a value of zero or lower prevents the buffer from expanding its size. A STREAM_MAXIMUM_SIZE_EXCEEDED error is raised when the buffer gets full. |  |
```

> | Max Buffer Size a| Number | This is the maximum amount of memory that will be used. If more than that is used then a STREAM_MAXIMUM_SIZE_EXCEEDED error will be raised. A value lower or equal to zero means no limit. |  |

Reviewer: ```suggestion
| Max Buffer Size a| Number | This is the maximum amount of memory that the buffer can use. If the buffer uses more than that, a STREAM_MAXIMUM_SIZE_EXCEEDED error is raised. A value lower or equal to zero means there's no limit. |  |
```

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required

> | Max In Memory Size a| Number | Defines the maximum memory that the stream should use to keep data in memory. If more than that is consumed then it will start to buffer the content on disk. |  |

Reviewer: ```suggestion
| Max In Memory Size a| Number | Defines the maximum memory that the stream can use to keep data in memory. If more than that is consumed, the content is buffered on disk. |  |
```

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Max Redelivery Count a| Number | The maximum number of times a message can be redelivered and processed unsuccessfully before triggering process-failed-message |  |
| Use Secure Hash a| Boolean | Whether to use a secure hash algorithm to identify a redelivered message |  |
| Message Digest Algorithm a| String | The secure hashing algorithm to use. If not set, the default is SHA-256. |  |
| Id Expression a| String | Defines one or more expressions to use to determine when a message has been redelivered. This property may only be set if useSecureHash is false. |  |

> | Object Store a| ObjectStore | The object store where the redelivery counter for each message is going to be stored. |  |

Reviewer: ```suggestion
| Object Store a| ObjectStore | The object store where the redelivery counter for each message is stored. |  |
```

| Field | Type | Description | Default Value | Required
| Host a| String | Host where the proxy requests will be sent. |  | x
| Port a| Number | Port where the proxy requests will be sent. |  | x
| Username a| String | The username to authenticate against the proxy. |  |
| Password a| String | The password to authenticate against the proxy. |  |

> | Non Proxy Hosts a| String | A list of comma separated hosts against which the proxy should not be used |  |

Reviewer: ```suggestion
| Non Proxy Hosts a| String | A list of comma separated hosts against which the proxy is not used |  |
```

| Ntlm Domain a| String | The domain to authenticate against the proxy. |  | x
| Host a| String | Host where the proxy requests will be sent. |  | x
| Port a| Number | Port where the proxy requests will be sent. |  | x
| Username a| String | The username to authenticate against the proxy. |  |
| Password a| String | The password to authenticate against the proxy. |  |

> | Non Proxy Hosts a| String | A list of comma separated hosts against which the proxy should not be used |  |

Reviewer: ```suggestion
| Non Proxy Hosts a| String | A list of comma separated hosts against which the proxy is not used |  |
```
